From 6783da7587ec42687a21696bacd21b808df60f50 Mon Sep 17 00:00:00 2001
From: Jiri Konecny <jkonecny@redhat.com>
Date: Thu, 21 Sep 2017 13:14:46 +0200
Subject: [PATCH 3/3] Fix missing id to name environment transition (#1491119)

Regression from Anaconda to Simpleline migration. Return method for
transition environment id to name which was removed during migration.

Resolves: rhbz#1491119
---
 pyanaconda/ui/tui/spokes/software_selection.py | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/pyanaconda/ui/tui/spokes/software_selection.py b/pyanaconda/ui/tui/spokes/software_selection.py
index 90de9e3a1..9d3aad0ce 100644
--- a/pyanaconda/ui/tui/spokes/software_selection.py
+++ b/pyanaconda/ui/tui/spokes/software_selection.py
@@ -117,7 +117,16 @@ class SoftwareSpoke(NormalTUISpoke):
     def _payload_error(self):
         self.errors = [payloadMgr.error]
 
-    def _get_environment_id(self, environment):
+    def _translate_env_selection_to_name(self, selection):
+        """ Return the selected environment name or None.
+            Selection can be None during kickstart installation.
+        """
+        if selection is not None:
+            return self.payload.environments[selection]
+        else:
+            return None
+
+    def _translate_env_name_to_id(self, environment):
         """ Return the id of the selected environment or None. """
         if environment is None:
             return None
@@ -250,8 +259,8 @@ class SoftwareSpoke(NormalTUISpoke):
                 # The environment was selected, switch screen
                 elif args is None:
                     # Get addons for the selected environment
-                    environment = self._selected_environment
-                    environment_id = self._get_environment_id(environment)
+                    environment = self._translate_env_selection_to_name(self._selected_environment)
+                    environment_id = self._translate_env_name_to_id(environment)
                     addons = self._get_available_addons(environment_id)
 
                     # Switch the screen
@@ -285,7 +294,7 @@ class SoftwareSpoke(NormalTUISpoke):
 
     def _apply(self):
         """ Private apply. """
-        self.environment = self._selected_environment
+        self.environment = self._translate_env_selection_to_name(self._selected_environment)
         self.addons = self._addons_selection if self.environment is not None else set()
 
         if self.environment is None:
@@ -305,7 +314,7 @@ class SoftwareSpoke(NormalTUISpoke):
                 self.data.packages.groupList = []
                 self.payload.selectEnvironment(self.environment)
 
-                environment_id = self._get_environment_id(self.environment)
+                environment_id = self._translate_env_name_to_id(self.environment)
                 available_addons = self._get_available_addons(environment_id)
 
                 for addon_id in available_addons:
-- 
2.13.5

